File: src/main/org/jboss/messaging/util/MessagingBuffer.java
/*
 * JBoss, Home of Professional Open Source
 * Copyright 2005-2008, Red Hat Middleware LLC, and individual contributors
 * by the @authors tag. See the copyright.txt in the distribution for a
 * full listing of individual contributors.
 *
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation; either version 2.1 of
 * the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */ 

package org.jboss.messaging.util;

/**
 * 
 * A MessagingBuffer
 * 
 * @author <a href="mailto:tim.fox@jboss.com">Tim Fox</a>
 *
 */
public interface MessagingBuffer
{
	void putByte(byte val);
	
	void putBytes(byte[] bytes);
	
	void putBytes(byte[] bytes, int offset, int length);
	
	void putInt(int val);
	
	void putInt(int pos, int val);
	
	void putLong(long val);
	
	void putShort(short val);
	
	void putDouble(double val);
	
	void putFloat(float val);
	
	void putBoolean(boolean val);
	
	void putChar(char val);
	
	void putNullableString(String val);
	
	void putString(String val);
			
	void putSimpleString(SimpleString val);
	
	void putNullableSimpleString(SimpleString val);
	
	void putUTF(String utf) throws Exception;
	
	byte getByte();
	
	short getUnsignedByte();
	
	void getBytes(byte[] bytes);
	
	void getBytes(byte[] bytes, int offset, int length);
	
	int getInt();
	
	long getLong();
	
	short getShort();
	
	int getUnsignedShort();
	
	double getDouble();
	
	float getFloat();
	
	boolean getBoolean();
	
	char getChar();
	
	String getString();
	
	String getNullableString();
	
	SimpleString getSimpleString();
	
	SimpleString getNullableSimpleString();
	
	String getUTF() throws Exception;
		
	byte[] array();
	
	int remaining();
	
	int capacity();
	
	int limit();
	
	void limit(int limit);
	
	void flip();
	
	void position(int position);
	
	int position();
	
	void rewind();
	
	MessagingBuffer slice();	
}

File: dotnet/JBMClient/remoting/spi/MessagingBuffer.cs
/*
 * JBoss, Home of Professional Open Source
 * Copyright 2005-2008, Red Hat Middleware LLC, and individual contributors
 * by the @authors tag. See the copyright.txt in the distribution for a
 * full listing of individual contributors.
 *
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation; either version 2.1 of
 * the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

namespace JBoss.JBM.Client.remoting.spi
{
    using JBoss.JBM.Client.util;

    /**
     *
     * A MessagingBuffer
     *
     * @author <a href="mailto:tim.fox@jboss.com">Tim Fox</a>
     *
     */
    public interface MessagingBuffer
    {
        void PutByte(byte val);

        void PutBytes(byte[] bytes);

        void PutBytes(byte[] bytes, int offset, int length);

        void PutInt(int val);

        void PutInt(int pos, int val);

        void PutLong(long val);

        void PutShort(short val);

        void PutDouble(double val);

        void PutFloat(float val);

        void PutBoolean(bool val);

        void PutChar(char val);

        void PutNullableString(string val);

        void PutString(string val);

        void PutSimpleString(SimpleString val);

        void PutNullableSimpleString(SimpleString val);

        void PutUTF(string utf);

        byte GetByte();

        short GetUnsignedByte();

        void GetBytes(byte[] bytes);

        void GetBytes(byte[] bytes, int offset, int length);

        int GetInt();

        long GetLong();

        short GetShort();

        int GetUnsignedShort();

        double GetDouble();

        float GetFloat();

        bool GetBoolean();

        char GetChar();

        string GetString();

        string GetNullableString();

        SimpleString GetSimpleString();

        SimpleString getNullableSimpleString();

        string GetUTF();

        byte[] ToArray();

        int Remaining();

        int Capacity();

        int Limit();

        void limit(int limit);

        void Flip();

        void Position(int position);

        int Position();

        void Rewind();

        MessagingBuffer Slice();

        MessagingBuffer CreateNewBuffer(int len);

        object GetUnderlyingBuffer();
    }
}